mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxView.kt

```diff
@@ -57,7 +68,7 @@ interface MvRxView : MvRxViewModelStoreOwner, LifecycleOwner {
      *
      * Default: [RedeliverOnStart].
      */
-    fun <S : MvRxState> BaseMvRxViewModel<S>.subscribe(deliveryMode: DeliveryMode = RedeliverOnStart, subscriber: (S) -> Unit) = subscribe(this@MvRxView, deliveryMode, subscriber)
+    fun <S : MvRxState> BaseMvRxViewModel<S>.subscribe(deliveryMode: DeliveryMode = RedeliverOnStart, subscriber: (S) -> Unit) = subscribe(this@MvRxView.subscriptionLifecycleOwner, deliveryMode, subscriber)
```

Comment:
it seems a little dangerous to continue to have MvRxView implement LifecycleOwner, since it can easily be passed instead of `subscriptionLifecycleOwner` on accident.

Since we're making breaking changes for 1.0, do you think we should consider changing this too?

Reply:
I thought about that (moving it as a property instead of an interface), but it seems like it would still be easy to use the wrong one.

However, almost all subscriptions should be going through the functions in MvRxView so I don't think there is too much risk.

Reply:
In debug mode, we could check if the view is an instance of Fragment and if the subscriptionLifecycleOwner == lifecycleOwner

Reply:
@gpeal The debug flag is only in the ViewModel, so no clean way to do this check in the Fragment. I tried to do it in the BaseMvrxViewModel, but a fragment may properly subscribe with its own or its view lifecycle owner depending on when the subscribe is called.

If someone makes a mistake in a custom MvRxView, it will result in a duplicate subscription, which we do throw an error for. I updated the error message to explain this possibility.